pkg/core/middleware/tenant.go

```diff
@@ -29,11 +30,15 @@ func (m TenantConfigurationMiddleware) Handle(next http.Handler) http.Handler {
 			panic(fmt.Errorf("Unable to retrieve configuration: %v", err.Error()))
 		}
 
+		r = auth.InitRequestAuthContext(r)
```

Comment:
We also have this line in `server/server.go`. Can we call this twice?

Reply:
In standalone mode, the init in `core/server` would override the init in `middleware/tenant`, i.e. the API key set in `middleware/tenant` would be ignored. However, `core/server` would resolve the API key in header, set by `middleware/tenant`, so it would still work correctly.

In a later PR (#1022), the authn part (i.e. resolving API key & session) would be refactored to be cleaner.

Reply:
Can we add the above note in the code?

Reply:
Added.